packages/server/src/sdk/app/oauth2/crud.ts

```diff
@@ -48,3 +48,46 @@ export async function get(id: string): Promise<OAuth2Config | undefined> {
   const doc = await getDocument()
   return doc?.configs?.[id]
 }
+
+export async function update(config: OAuth2Config): Promise<OAuth2Config> {
+  const db = context.getAppDB()
+  const doc: OAuth2Configs = (await getDocument(db)) ?? {
+    _id: DocumentType.OAUTH2_CONFIG,
+    configs: {},
+  }
+
+  if (!doc.configs[config.id]) {
+    throw new HTTPError(`OAuth2 config with id '${config.id}' not found.`, 404)
+  }
+
+  if (
+    Object.values(doc.configs).find(
+      c => c.name === config.name && c.id !== config.id
+    )
+  ) {
+    throw new HTTPError("Name is not available.", 400)
+  }
+
+  doc.configs[config.id] = {
+    ...config,
+  }
```

Comment:
Do we need to copy `config` here?

Reply:
Not really in here, but I will need it in further PRs for things like this without mutating the passed object:
![image](https://github.com/user-attachments/assets/0d2d3828-7cb4-46f8-bb23-f33b436bf6ea)